### PR TITLE
feat(scm): ignore scm data when scanning

### LINKED ISSUE #1251: Automatically ignore SCM data when scaning
### Description

When scanning an upload, I need to automatically ignore SCM specific data (Git, SVN, ... files).
Issue #703 hints at this, but also at ignoring other parts of the upload, so I would rather create a new issue concerning the SCM data specifically.

Not sure about the exact list of SCM solutions to be integrated (at least Git, SVN and TFS)

#### Steps followed and expected result

When uploading a new package for scanning (any type, really: from file, server, URL, SCM, TFS), a new option would enable the feature. See image below, where a new `4. Ignore SCM files` has been inserted.
When selected, the scan automatically discards all files related to known SCM solutions.

#### Screenshots

image

### PR DESCRIPTION
**Pull request by ChristopheRequillart**
### Description 
Implement ignore scm data when scanning an upload (see #1251)

### Changes
1. Add a new column scm in upload table
1. Add a new option to enable this feature when adding a new upload. 
1. Add a new function in ununpack/agent/utils.c to test SCM data file (see SCM_REGEX constant to search git, svn, TFS, mercurial).

## How to test
Add a new upload with scm data files. Enable this new option to not scanning SCM data files.

Closes #1251

### CODE REVIEW

File: src/www/ui/core-schema.dat [hunk 1]
```diff
@@ -1484,6 +1484,10 @@
   $Schema["TABLE"]["upload"]["public_perm"]["ADD"] = "ALTER TABLE \"upload\" ADD COLUMN \"public_perm\" int4";
   $Schema["TABLE"]["upload"]["public_perm"]["ALTER"] = "ALTER TABLE \"upload\" ALTER COLUMN \"public_perm\" DROP NOT NULL";
 
+  $Schema["TABLE"]["upload"]["scm"]["DESC"] = "COMMENT ON COLUMN \"upload\".\"scm\" IS 'true: automatically ignore SCM data when scaning, false: don\'t ignore it'";
```
Comment (GMishx) [inline src/www/ui/core-schema.dat]: Can you please fix the `don\'t`, it is failing in the Travis tests.
https://travis-ci.org/fossology/fossology/jobs/475316319#L2821

Comment (GMishx) [reply src/www/ui/core-schema.dat]: Also please try to rebase your branch with fossology master as there are some changes showing in your branch which are already merged by #1232

**Comment by GMishx** [comment]
The file `install/db/dbmigrate_scm.php` is not required as this line will perform the exact same job: https://github.com/fossology/fossology/pull/1264/files?utf8=%E2%9C%93&diff=unified&w=1#diff-7df9c36ae63f6333b821af1eeb1e8560R1488

File: src/www/ui/core-schema.dat (hunk 1 above)
Comment (ChristopheRequillart) [reply src/www/ui/core-schema.dat]: Thanks. OK I will update core-shema.dat
```suggestion
  $Schema["TABLE"]["upload"]["scm"]["DESC"] = "COMMENT ON COLUMN \"upload\".\"scm\" IS 'true: automatically ignore SCM data when scaning, false: dont ignore it'";
```
I don’t understand what changes done by me are already merged by #1232.

Comment (GMishx) [reply src/www/ui/core-schema.dat]: One of the changes: (Your PR is on the left and #1232 is on the right)

2019-01-07 17_09_35-virtualbox

Comment (ChristopheRequillart) [reply src/www/ui/core-schema.dat]: OK .travis.yml is now updated with fossology master version.

Comment (GMishx) [reply src/www/ui/core-schema.dat]: Hello @ChristopheRequillart, your PR still does not look in good shape. From the commits, I guess what you did was a merge of fossology:master to your branch.

Kindly rebase your branch to fossology:master. For help, please follow the [contributing guideline](https://github.com/fossology/fossology/blob/master/CONTRIBUTING.md#pull-requests).

Comment (ChristopheRequillart) [reply src/www/ui/core-schema.dat]: Hello @GMishx. I do a rebase as write in [contributing guideline](https://github.com/fossology/fossology/blob/master/CONTRIBUTING.md#pull-requests). But I have to do a merge to resolve diverging branches.

**Comment by GMishx** [comment]
Can you please rebase with origin master to resolve merge conflicts and squash your commits?

This branch have 13 commits and I think only 2894db401fb9596f2c71ce6c1cefd8719666e765 and 7deef0790229f7c0502f4b55252c432ce5d7da49 are required.

**Comment by GMishx** [comment]
Hello @ChristopheRequillart, can you please drop me an email at mishra.gaurav@siemens.com?

I need to discuss something.

**Comment by GMishx** [comment]
Shall we close this PR as it is continued on #1283 ?

**Comment by ChristopheRequillart** [comment]
Yes

**Pull request closed without merging**